options/glibc/generic/getopt.cpp

```diff
@@ -237,9 +222,66 @@ int getopt_common(int argc, char * const argv[], const char *optstring, const st
 			}
 		}
 	}
+
+	optarg = nullptr;
 	return -1;
 }
 
+void permute(char **argv, int dest, int src) {
```

Comment:
Either assert that `src > dest` and add a comment, or simply handle the case where `src <= dest`.

Reply:
The assert was added, after which an OK from @avdgrinten was received.